### DISCUSSION TITLE
Consistency in name fields

### DISCUSSION THREAD

**Discussion by glennjacobs** [lunarphp/lunar#1234] [Roadmap]
Currently, Brand and Customer models have specific name fields, which are sometimes translatable fields.

However, Products and Collections have their names specified within the custom attributes. Ideally, we would make this more consistent so that Products and Collections have a specific `name` field in the database. This would allow the system to rely upon the name fields more, as currently a developer could remove these attributes and cause errors in the system.

**Comment by netzknecht** [comment]
I've also noticed the different approaches. I also think it makes sense to take a uniform approach. 

In my project, I converted the `ProductType` model attribute into a JSON field and made it compatible for translations using an `AsTranslation` cast instead of extending the `ProductType` model with the Lunar attributes.

But with regard to uniformity, the requirement of internationalization and to take the advantages of Lunar attributes into account, it makes the most sense to map everything that needs to be translated as Lunar attributes and not as model attributes / database fields.

In this regard, it also makes sense to move the `attribute_data` database field into its own schema and use its own model with polymorphic relationships to models, makes use of it.

Ultimately, another term like “Fields” instead of “Attributes” seems useful to me. The term "Fields" is already partly used in connection with the Lunar attributes and would avoid misunderstandings and naming conflicts with the model attributes.

When using JSON fields, I noticed that PHP and MySQL serialize and deserialize JSON data differently. In conjunction with Laravel, this leads to a bug that has not yet been fixed, which always defines models as "changed", which is sometimes problematic. The error is reported here, for example: https://github.com/laravel/framework/issues/46247

One solution from Laravel would be to emulate JSON serialization depending on the database driver or to do without JSON fields, which would actually be a shame.

**Reply by netzknecht** [reply]
> When using JSON fields, I noticed that PHP and MySQL serialize and deserialize JSON data differently. In conjunction with Laravel, this leads to a bug that has not yet been fixed, which always defines models as "changed", which is sometimes problematic. The error is reported here, for example: https://github.com/laravel/framework/issues/46247

My first approach to fix it: https://github.com/laravel/framework/compare/10.x...netzknecht:laravel-framework:bugfix-issue-46247
